### PR TITLE
Shimmer: refactor in preparation for migration to OUFR

### PR DESCRIPTION
**Pull request by Vitalius1**
#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [X] Include a change request file using `$ npm run change`

#### Description of changes
This PR is the last refactor of Shimmer and this version is intended for migration to OUFR. Most of the changes are moving stuff around so functionally nothing changed.

[Shimmer](https://odsp-int.azurewebsites.net/v-vibr/Shimmer) - Example. (give it a couple seconds to load :bowtie:)

Things done:
- [X] Support for RTL
- [X] Support for HighContrast
- [X] Support for screen-readers
- [X] Support for theming
- [X] Built following mergeStyles pattern
- [X] Modularized as much as possible
- [X] Support for different browser (IE11, Edge, Firefox, Chrome, Safari, Opera)
- [X] A lot of examples on using Shimmer for building various skeletons of UI
- [X] Deprecated props are warned in the console

### CODE REVIEW

File: packages/experiments/src/components/Shimmer/Shimmer.base.tsx [hunk 1]
```diff
@@ -1,161 +1,136 @@
...
+    } else {
+      this.setState({
+        contentLoaded: isDataLoaded
+      });
+    }
   }
 
   public render(): JSX.Element {
     const {
       getStyles,
```
Comment (jordandrako) [inline packages/experiments/src/components/Shimmer/Shimmer.base.tsx]: Approved, but `getStyles` is being replaced with `styles` in 6.0. Depending on when you want this merged, before or after 6.0 is merged to master, you may need to replace getStyles in the props with styles. The function `getStyles` within the .styles.ts file can remain `getStyles`.

Comment (jordandrako) [reply packages/experiments/src/components/Shimmer/Shimmer.base.tsx]: See this commit for details: https://github.com/OfficeDev/office-ui-fabric-react/commit/f226fb413dc5d88a36092b9e70b081313c5da37e

**Comment by Vitalius1** [comment]
@dzearing Do you know why bundlesize check is still pending? Also, is there anyone from your team available for a code review? Thanks!

File: packages/experiments/src/components/Shimmer/ShimmerElementsGroup/ShimmerElementsGroup.types.ts [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  /**
+   * Elements to render in one group of the Shimmer.
+   */
+  shimmerElements?: IShimmerElement[];
+
+  /**
+   * Optional boolean for enabling flexWrap of the container containing the shimmerElements.
+   * @default false
+   */
+  flexWrap?: boolean;
```
Comment (atneik) [inline packages/experiments/src/components/Shimmer/ShimmerElementsGroup/ShimmerElementsGroup.types.ts]: Should this be `true` by default?

Comment (Vitalius1) [reply packages/experiments/src/components/Shimmer/ShimmerElementsGroup/ShimmerElementsGroup.types.ts]: No. This subcomponent is also used internally by ShimmerBase when everything stays in one line. So by default is false.

File: packages/experiments/src/components/Shimmer/Shimmer.types.ts [hunk 3]
```diff
@@ -140,41 +172,98 @@ export interface IGap extends IShimmerElement {
...
+  /**
+   * @deprecated Use 'bottom' instead
+   */
+  BOTTOM = 2,
+
+  /**
+   * @deprecated Use 'top' instead
+   */
+  TOP = 3,
+
```
Comment (JasonGore) [inline packages/experiments/src/components/Shimmer/Shimmer.types.ts]: No need to support deprecated values

File: packages/experiments/src/components/Shimmer/Shimmer.types.ts [hunk 4]
```diff
@@ -43,30 +35,70 @@ export interface IShimmerProps extends React.AllHTMLAttributes<HTMLElement> {
...
+   */
+  width?: number;
+
+  /**
+   * Use when providing custom skeleton as children wrapped by shimmer.
+   * Deprecated in favor of 'customElementsGroup'
+   * @default false
+   * @deprecated Use 'customElementsGroup' instead.
+   */
+  isBaseStyle?: boolean;
```
Comment (JasonGore) [inline packages/experiments/src/components/Shimmer/Shimmer.types.ts]: Please remove deprecated props

File: packages/experiments/src/components/Shimmer/Shimmer.base.tsx [hunk 5]
```diff
@@ -1,161 +1,136 @@
...
+    this._warnDeprecations({
+      'isBaseStyle': 'customElementsGroup',
+      'width': 'widthInPercentage or widthInPixel',
+      'lineElements': 'shimmerElements'
+    });
+
+    this._warnMutuallyExclusive({
+      'lineElements': 'shimmerElements',
+      'customElementsGroup': 'lineElements'
+    });
```
[the remark this answers is not in the document]
Comment (Vitalius1) [reply packages/experiments/src/components/Shimmer/Shimmer.base.tsx]: @JasonGore  This experimental component is used in odsp-common repo in items-view and odsp-shared-react packages. So removing the deprecated props will cause a break on their side. The intention is to merge this change and right after I will migrate this component to OUFR without the deprecated props. Next step will be to go and change in odsp-common the imports from experiments to office-ui-fabric-react and use the correct props. Only when all this happens I will be able to remove the deprecated props in experiments package without breaking the contract with Shimmer API.

File: packages/experiments/src/components/Shimmer/Shimmer.types.ts (hunk 3 above)
Comment (Vitalius1) [reply packages/experiments/src/components/Shimmer/Shimmer.types.ts]: same as above

File: packages/experiments/src/components/Shimmer/Shimmer.types.ts (hunk 4 above)
Comment (Vitalius1) [reply packages/experiments/src/components/Shimmer/Shimmer.types.ts]: same as above

File: packages/experiments/src/components/Shimmer/Shimmer.base.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (JasonGore) [reply packages/experiments/src/components/Shimmer/Shimmer.base.tsx]: Ok, thanks for the explanation

**Pull request merged**